### PR TITLE
Bounds animation

### PR DESCRIPTION
**Pull request by adamjernst**
Bounds animation now apply to subview bounds, not just top-level collection view changes.

### CODE REVIEW

File: ComponentKit/Core/ComponentViewManager.mm [hunk 1]
```diff
@@ -53,6 +53,13 @@ void addString(const std::string &s)
         return identifiers == k.identifiers;
       }
     };
+
+    struct ActionDisabler {
+      ActionDisabler() : _originalValue([CATransaction disableActions]) { [CATransaction setDisableActions:YES]; }
+      ~ActionDisabler() { [CATransaction setDisableActions:_originalValue]; }
```
Comment (ocrickard) [inline ComponentKit/Core/ComponentViewManager.mm]: yay not forgetting to reset action settings!

File: ComponentKit/Core/CKComponentBoundsAnimation.h [hunk 2]
```diff
@@ -45,3 +45,13 @@ struct CKComponentBoundsAnimation {
...
   CGFloat springInitialVelocity;
 };
+
+/**
+ Wraps the given block in the correct UIView animation block for a given bounds animation.
+ If duration is zero, wraps [UIView +performWithoutAnimation:].
+ If mode is Default, wraps [UIView +animateWithDuration:...].
+ If mode is Spring, wraps [UIView +animateWithDuration:delay:usingSpringWithDamping:...]
+ */
+void CKComponentBoundsAnimationApply(const CKComponentBoundsAnimation &animation,
```
Comment (modocache) [inline ComponentKit/Core/CKComponentBoundsAnimation.h]: I'm loving the `CKComponentBoundsAnimation` struct. :heart_eyes:

File: ComponentKit/DataSources/CKComponentBoundsAnimation+UICollectionView.mm [hunk 3]
```diff
@@ -156,21 +156,7 @@ - (void)applyBoundsAnimationToCollectionView:(const CKComponentBoundsAnimation &
...
-                     animations:restore
-                     completion:completion];
-  } else {
-    [UIView animateWithDuration:animation.duration
-                          delay:animation.delay
-                        options:0
-                     animations:restore
-                     completion:completion];
-  }
+  CKComponentBoundsAnimationApply(animation, restore, completion);
```
Comment (ocrickard) [inline ComponentKit/DataSources/CKComponentBoundsAnimation+UICollectionView.mm]: Nice.

**Comment by ocrickard** [comment]
Looks good to me!

**Pull request merged**